Library/Homebrew/system_config.rb

```diff
@@ -0,0 +1,144 @@
+    end
+
+    def core_tap_head
+      CoreTap.instance.git_head || "(none)"
+    end
+
+    def core_tap_last_commit
+      CoreTap.instance.git_last_commit || "never"
+    end
+
+    def core_tap_origin
+      CoreTap.instance.remote || "(none)"
+    end
+
+    def describe_path(path)
+      return "N/A" if path.nil?
+      realpath = path.realpath
+      if realpath == path
+        path
+      else
+        "#{path} => #{realpath}"
+      end
+    end
+
+    def describe_perl
+      describe_path(which "perl")
+    end
+
+    def describe_python
+      python = which "python"
+      return "N/A" if python.nil?
+      python_binary = Utils.popen_read python, "-c", "import sys; sys.stdout.write(sys.executable)"
+      python_binary = Pathname.new(python_binary).realpath
+      if python == python_binary
+        python
+      else
+        "#{python} => #{python_binary}"
+      end
+    end
+
+    def describe_ruby
+      ruby = which "ruby"
+      return "N/A" if ruby.nil?
+      ruby_binary = Utils.popen_read ruby, "-rrbconfig", "-e", \
+        'include RbConfig;print"#{CONFIG["bindir"]}/#{CONFIG["ruby_install_name"]}#{CONFIG["EXEEXT"]}"'
+      ruby_binary = Pathname.new(ruby_binary).realpath
+      if ruby == ruby_binary
+        ruby
+      else
+        "#{ruby} => #{ruby_binary}"
+      end
+    end
+
+    def hardware
+      return if Hardware::CPU.type == :dunno
+      "CPU: #{Hardware.cores_as_words}-core #{Hardware::CPU.bits}-bit #{Hardware::CPU.family}"
+    end
+
+    def kernel
+      `uname -m`.chomp
+    end
+
+    def describe_java
+      # java_home doesn't exist on all OS Xs; it might be missing on older versions.
+      return "N/A" unless File.executable? "/usr/libexec/java_home"
+
+      java_xml = Utils.popen_read("/usr/libexec/java_home", "--xml", "--failfast")
+      return "N/A" unless $?.success?
+      javas = []
+      REXML::XPath.each(REXML::Document.new(java_xml), "//key[text()='JVMVersion']/following-sibling::string") do |item|
+        javas << item.text
+      end
+      javas.uniq.join(", ")
+    end
+
+    def dump_verbose_config(f = $stdout)
+      f.puts "HOMEBREW_VERSION: #{HOMEBREW_VERSION}"
+      f.puts "ORIGIN: #{origin}"
+      f.puts "HEAD: #{head}"
+      f.puts "Last commit: #{last_commit}"
+      if CoreTap.instance.installed?
+        f.puts "Core tap ORIGIN: #{core_tap_origin}"
+        f.puts "Core tap HEAD: #{core_tap_head}"
+        f.puts "Core tap last commit: #{core_tap_last_commit}"
+      else
+        f.puts "Core tap: N/A"
+      end
+      f.puts "HOMEBREW_PREFIX: #{HOMEBREW_PREFIX}"
+      f.puts "HOMEBREW_REPOSITORY: #{HOMEBREW_REPOSITORY}"
+      f.puts "HOMEBREW_CELLAR: #{HOMEBREW_CELLAR}"
+      f.puts "HOMEBREW_BOTTLE_DOMAIN: #{BottleSpecification::DEFAULT_DOMAIN}"
+      f.puts hardware if hardware
+      f.puts "GCC-4.0: build #{gcc_40}" if gcc_40
+      f.puts "GCC-4.2: build #{gcc_42}" if gcc_42
+      f.puts "LLVM-GCC: build #{llvm}"  if llvm
```

Comment:
Yes, I guessed so and am totally fine with that. (Hence my last sentence above.)